Initial commit of acceptance tests

jamesls opened the pull request:
These add the basic smoke tests written in
gherkin.  These are the same shared tests across
the AWS SDKs.

I think the eventual plan is to remove most the test_smoke.py
tests in favor of these tests.

For now, I've added the tests so we can run the manually until
we're ready to make the switch.

The tests are all passing.

To run these:

1. Install behave (`pip install -r requirements.txt`)
2. `cd tests/acceptance`
3. Run: `behave`.

The two files that implement the logic for running the tests are `environment.py` and `steps/base.py`.

More info on the test lib: https://pythonhosted.org/behave/index.html

cc @kyleknap @mtdowling @rayluo @JordonPhillips gg

JordonPhillips commented:
I'm getting some failures:

```
@cloudfront
Feature: Amazon CloudFront # features/smoke/cloudfront/cloudfront.feature:3

  Scenario: Making a basic request                              # features/smoke/cloudfront/cloudfront.feature:5
    When I call the "ListDistributions" API with                # features/steps/base.py:45 0.383s
      | MaxItems | 1 |
    Then the value at "DistributionList.Items" should be a list # features/steps/base.py:80 0.001s
      Assertion Failed: None is not an instance of <type 'list'>
      Captured logging:
      INFO:botocore.vendored.requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): cloudfront.amazonaws.com


...

@sqs
Feature: Amazon Simple Queue Service # features/smoke/sqs/sqs.feature:3

  Scenario: Making a request                       # features/smoke/sqs/sqs.feature:5
    When I call the "ListQueues" API               # features/steps/base.py:40 0.372s
    Then the value at "QueueUrls" should be a list # features/steps/base.py:80 0.000s
      Assertion Failed: None is not an instance of <type 'list'>
      Captured logging:
      INFO:botocore.vendored.requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): queue.amazonaws.com
```

jamesls commented:
@JordonPhillips ah that makes sense.  If there are no elements the key does not exist.  What would guys think about making "is a list" a noop?  I personally prefer that we just verify we can a success response back.  Otherwise we run into issues like this.

JordonPhillips commented:
Yeah I think a success response is probably good here.

kyleknap commented:
Looks fine. :ship: on the implementation. Let's get this test fixed that @JordonPhillips noted before merging. Also could we get a ci script to run the acceptance tests like we do for ``run-tests`` and ``run-integ-tests``? Would be useful when we hook this test suite up to some of our jobs.

jamesls commented:
@kyleknap Do you want that as part of this PR?

jamesls commented:
@kyleknap Nevermind, I just added it to this PR.  Also fixed the issue with the list type checking step.

Ready to be looked at again (cc @JordonPhillips).

kyleknap commented:
Cool thanks!

JordonPhillips commented:
:ship:
